woof-code/rootfs-petbuilds/palemoon/pinstall.sh

```diff
@@ -1,3 +1,19 @@
-echo '#!/bin/sh
-exec palemoon "$@"' > usr/local/bin/defaultbrowser
-chmod 755 usr/local/bin/defaultbrowser
+# Set defaultbrowser to run x-www-browser - set the default browser the Debian way, not the Puppy way.
+case ${DISTRO_BINARY_COMPAT} in
+    ubuntu|trisquel|debian|devuan|raspbian)
+        /usr/bin/update-alternatives --install /usr/bin/x-www-browser x-www-browser /usr/bin/palemoon 200
+        echo '#!/bin/sh
+CMD=$(command -v x-www-browser)
```

Comment:
You'll need to use `chroot . sh -c 'command -v x-www-browser'`, because the shell that runs pinstall.sh runs on the host system and not what you're building (i.e. you're checking if the OS you're running woof-CE on has x-www-browser, not the OS you're building)

Reply:
Please note this line is wrapped within single quotes. So, this is the content of the file `/usr/local/bin/defaultbrowser`, not actually run in the `pinstall.sh` file.

Reply:
Although, I am not really sure if this check is needed. By what I understood by the code you referred to, I understood that `defaultbrowser` always makes this check.

Reply:
> not actually run in the pinstall.sh file.

You're right, I admit I reviewed this very quickly while also trying to make a living :)

Reply:
The situation in Puppy is a bit weird because when browsers ask to set themselves as the default they do so through xdg-mime and mimeapps.list: they basically override Puppy's hack to set defaultbrowser.desktop as the default browser. You end up with two default browsers: one that's the default in ROX-Filer and Puppy's replacement for xdg-open, and one for everything else.

Reply:
> one that's the default in ROX-Filer and Puppy's replacement for xdg-open, and one for everything else

I think that although it is not a desired behaviour, still when I use Puppy, I do not mind going to "Set default apps" application and setting the new default (for example if I installed a new browser).

Reply:
The "Set default apps" changes defaultbrowser and friends instead of changing mimeapps.list (which sets defaultbrowser as the default for https:// URLs, etc'). In contrast, the "set as default" feature of browsers changes mimeapps.list, while root and spot have separate mimapps.list files.

Reply:
https://github.com/puppylinux-woof-CE/woof-CE/blob/4c8a372cd9bec68d687d2f8018e7121d5563efca/woof-code/3builddistro#L1030

This block in 3builddistro makes Puppy behave pretty much like other distros by 1) removing Puppy's mimeapps.list, which sets defaultbrowser as the handler for https:// URLs, etc' 2) making defaultbrowser run the default handler for https:// URLs instead of the other way around (if the user sets the default browser through the browser or through the file manager, defaultbrowser respects this decision).

This is activated when you build a Puppy without ROX-Filer, because defaultbrowser, Puppy's mimeapps.list and other Puppy-specific hacks exist only to make ROX-Filer happy.

I don't see a clean way to make the browser's "set as default" feature work in a Puppy with ROX-Filer, unless you're willing to make very significant changes like removing Puppy's mimeapps.list and xdg-open.